sdk/jobs/E2E-ds-experience/e2e-ml-workflow-part-I.ipynb

```diff
@@ -0,0 +1,1225 @@
+    "- writing directly the yaml specification of the component or create it programmatically using `ComponentMethod`,\n",
+    "- optionally register this component with a name and version in your workspace to make it reusable and shareable,\n",
+    "- load that component from the pipeline code\n",
+    "- implement the pipeline using this component inputs, outputs and parameters\n",
+    "- submit the pipeline."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "34e89b69",
+   "metadata": {},
+   "source": [
+    "## 3.1. Create or Load Components\n",
+    "\n",
+    "### 3.1.1 Data Preparation Component (using programmatic definition)\n",
+    "\n",
+    "Let's start by creating the first component. This component handles the preprocessing of the data. The preprocessing task is performed in the *data_prep.py* python file.\n",
+    "\n",
+    "Let's first create a source folder for the data_prep component:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0d1cb49a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "data_prep_src_dir = \"./components/data_prep\"\n",
+    "os.makedirs(data_prep_src_dir, exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b0e39f3",
+   "metadata": {},
+   "source": [
+    "This script performs the simple task of splitting the data into train and test datasets. \n",
+    "Azure ML mounts datasets as folders to the computes, therefore, we created an auxiliary `select_first_file` function to access the data file inside the mounted input folder.\n",
+    "\n",
+    "[MLFlow](https://mlflow.org/docs/latest/tracking.html) can be used to log the parameters and metrics during our pipeline run. A detailed guide on Azure ML logging is available [here](https://github.com/Azure/azureml-examples/blob/sdk-preview/notebooks/mlflow/mlflow-v1-comparison.ipynb). \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d0e23e3f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile {data_prep_src_dir}/data_prep.py\n",
+    "import os\n",
+    "import argparse\n",
+    "import pandas as pd\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "import logging\n",
+    "from azureml.core import Run\n",
+    "import mlflow\n",
+    "\n",
+    "\n",
+    "def select_first_file(path):\n",
+    "    \"\"\"Selects first file in folder, use under assumption there is only one file in folder\n",
+    "    Args:\n",
+    "        path (str): path to directory or file to choose\n",
+    "    Returns:\n",
+    "        str: full path of selected file\n",
+    "    \"\"\"\n",
+    "    files = os.listdir(path)\n",
+    "    return os.path.join(path, files[0])\n",
+    "\n",
+    "\n",
+    "# input and output arguments\n",
+    "parser = argparse.ArgumentParser()\n",
```

Comment:
Nit: suggest putting this logic in a `main()` method you call from an `if __name__ == "__main__"` block.

Reply:
Since I am writing the code inside the notebook cell, I prefer to minimize all boiler plate code. Basically, unlike some other samples, I am not suggesting the component sources are prod level.

Reply:
@AliMahmoudzadeh , I don't see my recommendation as prod-level or boilerplate, it's a very standard way of organizing Python code.